contracts/LoopToken.sol

```diff
@@ -1,10 +1,69 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.15;
 
-import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+// import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 
-contract LoopToken is ERC20 {
-    constructor(uint256 initialSupply) ERC20("LoopToken", "LT") {
-        _mint(msg.sender, initialSupply);
+// contract LoopToken is ERC20 {
+//     constructor(uint256 initialSupply) ERC20("LoopToken", "LT") {
+//         _mint(msg.sender, initialSupply);
+//     }
+// }
+
+contract LoopToken {
+    mapping(address => uint256) private _balances;
+
+    string private _name;
+    string private _symbol;
+    uint8 private _decimals;
+    uint256 private _totalSupply;
+
+    event Transfer(address indexed from, address indexed to, uint256 amount);
+
+
+    constructor(string memory name_, string memory symbol_, uint8 decimals_, uint256 totalSupply_) {
```

Comment:
Sobre tu consulta sobre el totalSupply y la cantidad de decimales: 

- Decimales: podemos asumir 18 como default, todos o casi todos los tokens mantienen 18 decimales que es el máximo admitido por la EVM
- Totalsupply: hay un buen comentario en ERC20.sol de openzeppelin 

"This implementation is agnostic to the way tokens are created. This means
  that a supply mechanism has to be added in a derived contract using {_mint}."

Basicamente provee una función _mint que "genera" el balance, el total supply. 

```
function _mint(address account, uint256 amount) internal virtual {
        require(account != address(0), "ERC20: mint to the zero address");
        _totalSupply += amount;
         _balances[account] += amount;
        emit Transfer(address(0), account, amount);
    }
```

Podrías llamar a esta función desde el constructor o directamente implementarlo en el constructor. 

Tiene sentido tener otra función _mint separada del constructor si tenemos casos donde el contrato inicia con un supply X y con el tiempo se genera mayor supply (seria un token de supply variable)